### PR TITLE
Changed .count and .length for .size in plots2/main/app/models/node.rb file

### LINKED ISSUE #8204: Replace .count and .length in node model with .size
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file https://github.com/publiclab/plots2/blob/main/app/models/node.rb , replace all instances of `.count` and `.length` with `.size`
[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
$DIFF
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it. 

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below!

### PR DESCRIPTION
**Pull request by lax0248917**
Fixes #8204  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by lax0248917** [comment]
@publiclab/reviewers

Received this error after submitting a pull request. Not sure if this is a stop to this merge. It is unclear if the issue was fixed. Any advice to resolve this issue would be greatly appreciated.
image

**Comment by gauravano** [comment]
**Some suggestion on PR structuring:**

1. Naming branches:
* Every branch can be grouped in one of the following buckets: `feature/`, `refactor/`, `bugfix/` , `hotfix/`
* Give a very brief but descriptive enough name to the branch like `bugfix/question-controller` rather than naming branch wrt it's respective ticket
Good branch names are as important and helpful as good commit messages in situations we have to do a quick revert/hotfix.

Also on a side note, as a convention, commit messages should be in the present tense (what is the implication of this particular commit), example: changes node model to use .size for improving query performance

**Comment by lax0248917** [comment]
@gauravano Thank you for your response. To follow up with branch naming, is it language specific? I have mostly worked with HTML, CSS, and JavaScript and have followed this style of branch naming feature_branch_name or feature-branch-name. I have not seen it done as you mentioned above. But, that being said, for this issue a better name would have been something like refactor/change-method-names?

Also, I will start troubleshooting why that test failed and see what I can come up with. Thanks again.

**Comment by gauravano** [comment]
> @gauravano Thank you for your response. To follow up with branch naming, is it language specific? I have mostly worked with HTML, CSS, and JavaScript and have followed this style of branch naming feature_branch_name or feature-branch-name. I have not seen it done as you mentioned above. But, that being said, for this issue a better name would have been something like refactor/change-method-names?

No, it's not language-specific. It's just a way to categorize Git branches just like we do categorize files and folders on our desktop. Suppose, you're looking for your old branches for some work, in that case, you can shorten your search bucket by looking for a specific keyword in the branch name like feature, refactor, bugfix, etc. It also kind of put a tag on your branches if it's feature branch or what. 

> Also, I will start troubleshooting why that test failed and see what I can come up with. Thanks again.

Sure.

File: app/models/node.rb [hunk 1]
```diff
@@ -187,14 +187,14 @@ def self.weekly_tallies(type = 'note', span = 52, time = Time.now)
...
-                               .count
+                               .size
     end
     weeks
   end
 
   def self.contribution_graph_making(type = 'note', start = Time.now - 1.year, fin = Time.now)
     date_hash = {}
-    week = start.to_date.step(fin.to_date, 7).count
+    week = start.to_date.step(fin.to_date, 7).size
```
Comment (cesswairimu) [inline app/models/node.rb:197]: Hi, thanks for working on this :100:  lets leave this as `count`...using size here will return nil

File: app/models/node.rb [hunk 2]
```diff
@@ -214,7 +214,7 @@ def self.frequency(type, starting, ending)
     weeks = (ending.to_date - starting.to_date).to_i / 7.0
     Node.published.select(%i(created type))
       .where(type: type, created: starting.to_i..ending.to_i)
-      .count(:all) / weeks
+      .size(:all) / weeks
```
Comment (cesswairimu) [inline app/models/node.rb:217]: `size(:all)`  is not a valid method ...lets remove `:all` here and leave it as `.size / weeks` we should get the correct value with that.

File: app/models/node.rb (hunk 1 above)
Comment (lax0248917) [reply app/models/node.rb]: Thank you for the help. Changes made in latest commit

File: app/models/node.rb (hunk 2 above)
Comment (lax0248917) [reply app/models/node.rb]: Thank you for the advice. Changes made in the latest commit.

**Comment by lax0248917** [comment]
Looks like the two above suggested code changes were enough to have the Travis CI test pass. I don't believe I am going to be able to get the codeclimate test to pass with all the nested control flow statements.

**Comment by cesswairimu** [comment]
Thanks @lax0248917 for the update...no worries about codeclimate...I will help with that

**Review by cesswairimu — APPROVED** [review]

**Comment by cesswairimu** [comment]
@jywarren, I think this is as much refactor we can do on the method as of now...kindly take a look at the codeclimate issue

**Comment by jywarren** [comment]
Hi @cesswairimu that sounds great, i'm sorry i missed this, but merging it now! Thanks, everyone!!!! 🎉

**Comment by cesswairimu** [comment]
Thanks :+1:

**Pull request merged**